### ISSUE TITLE
Color management (of tags)

### ISSUE THREAD

**Issue by pzl** [danielquinn/paperless#284]
Putting this here to gauge interest in the following change to tag-color behavior:

In the various tag-editor pages (creation and editing) change the color dropdowns to be a color-picker, and allow arbitrary color values, instead of a range of a few predetermined ones.

This seems like it would mostly involve changing the colour field from what's effectively an Enum to be a Char field, and use `<input type=color>` for it's interface.

A migration path would be created for changing values like `2` or `5` from older versions into their respective color strings. And then the forward-going breaking change would be the API response. So API clients (e.g. paperless-desktop) would now have color-string responses (a-la "#e31a1c") instead of the numeric 6. Not sure how you would want to handle that last part.

**Comment by danielquinn** [comment]
Yeah I wanted to do this when I first wrote that section, but didn't have the time/tools to do it right.  Since then, I've discovered [django-colorfield](https://github.com/jaredly/django-colorfield) which might do a lot of the work for us.

I'll tag this one as a feature, and see to it when I have some time.  Should you fancy doing the work yourself though, feel free to send me a PR :-)

**Comment by mikkelnl** [comment]
Sorry to kick this old topic, but I was wondering if it is possible somehow to add colors for tags? I only have the few default ones available and would like to add some extra new colors...

**Comment by jat255** [comment]
@mikkelnl I haven't tried it, but it looks like those colors are defined in https://github.com/the-paperless-project/paperless/blob/master/src/documents/migrations/0007_auto_20160126_2114.py#L28

Perhaps you could try adding some colors to that list, and then rerun the migrations? Not sure that's the best way, or if it would even work...

**Comment by mikkelnl** [comment]
> @mikkelnl I haven't tried it, but it looks like those colors are defined in https://github.com/the-paperless-project/paperless/blob/master/src/documents/migrations/0007_auto_20160126_2114.py#L28
> 
> Perhaps you could try adding some colors to that list, and then rerun the migrations? Not sure that's the best way, or if it would even work...

Thanks, but I have no idea how/where to edit this file ;)
